packages/flutter_tools/lib/src/commands/emulators.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/common.dart';
+import '../base/platform.dart';
+import '../base/utils.dart';
+import '../doctor.dart';
+import '../emulator.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+
+class EmulatorsCommand extends FlutterCommand {
+  EmulatorsCommand() {
+    argParser.addOption('start',
+        help: 'The full or partial ID of the emulator to start.');
+  }
+
+  @override
+  final String name = 'emulators';
+
+  @override
+  final String description = 'List all available emulators.';
+
+  @override
+  Future<Null> runCommand() async {
+    if (doctor.workflows.every((Workflow w) => !w.canListEmulators)) {
+      throwToolExit(
+          'Unable to find any emulator sources. Please ensure you have some\n'
+          'Android AVD images ' + (platform.isMacOS ? 'or an iOS Simulator ' : '')
+          + 'available.',
+          exitCode: 1);
+    }
+
+    if (argResults.wasParsed('start')) {
+      await _startEmulator(argResults['start']);
+    } else {
+      await _listEmulators();
+    }
+  }
+
+  Future<Null> _startEmulator(String id) async {
+    final List<Emulator> emulators =
+        await emulatorManager.getEmulatorsById(id).toList();
+
+    if (emulators.isEmpty) {
+      printStatus("No emulator found that matches the ID '$id'.");
+    } else if (emulators.length > 1) {
+      printStatus("More than one emulator matches the ID '$id':\n");
+      Emulator.printEmulators(emulators);
+    } else {
+      emulators.first.launch();
```

Comment:
`await` this?

Reply:
Ugh, looks like there was a bug here. When I await it, the process never terminates until the emulator is closed. I'm not sure how to fix this - we have a `runDetached` method but then we wouldn't get any output. Any suggestions?